src/rcms/fm/app/level1/HCALEventHandler.java

```diff
@@ -2531,7 +2535,20 @@ public void run() {
                 // total status not OK and FMstate != RunningDegraded => go to degraded state 
                 if(!FMstate.equals(HCALStates.RUNNINGDEGRADED.toString())) {
                   logger.warn("[HCAL " + functionManager.FMname + "] AlarmerWatchThread: due to bad alarmer status (see previous messages), going to RUNNINGDEGRADED state");
-                  functionManager.fireEvent(HCALInputs.SETRUNNINGDEGRADED);
+                  String runningDegradedReason = "Running degraded due to";
+                  for (String partitionName : badAlarmerPartitions){
+                    if (ignoredPartitions.contains(partitionName)){
+                      continue;
+                    }
+                    String alarmDetails  = pam.getValue(partitionName+"_Message");
+                    if (alarmDetails!=null){
+                      runningDegradedReason += " " + partitionName + ":" + alarmDetails " |";
```

Comment:
I apologize I forgot I had made changes to the error message since the last test. I will fix this shortly.